### ISSUE TITLE
Can only get 4 properties?

### ISSUE THREAD

**Issue by nikagersonlohman** [android/car-samples#12]
Any thoughts why I can only get 4 properties?

CarPropertyConfig{mPropertyId= 287310850
CarPropertyConfig{mPropertyId= 289408000
CarPropertyConfig{mPropertyId= 289408001
CarPropertyConfig{mPropertyId= 289408009

When I list all properties, I only see the above 4:

```
        // Create a list of properties in debug window
        val carPropertyList = carPropertyManager.propertyList
        Log.d(TAG, "carPropertyList: " + carPropertyList.toString())
        for (i in 0 until carPropertyList.size) {
            val carProperty = carPropertyList[i]
            data.add(carProperty.toString())
        }
```

If I try to register any other property, I get an error:
E/CarPropertyManager: registerListener:  propId is not in config list:  291504647

Here's the code I try to use:
```
        // Subscribes to the car speed events.
        carPropertyManager.registerCallback(
            carPropertyCarSpeedListener,
            VehiclePropertyIds.PERF_VEHICLE_SPEED,
            CarPropertyManager.SENSOR_RATE_ONCHANGE
        )
```

I did add CAR_SPEED permission to my manifest:
```
    <uses-permission android:name="android.car.permission.CAR_SPEED" />
```

**Comment by frankkienl** [comment]
Maybe some other permissions are needed as well 🤔

**Comment by nikagersonlohman** [comment]
Yeah... but which... who knows... where to ask?

**Comment by asimwaleed** [comment]
i also got this error .

**Comment by nikagersonlohman** [comment]
I have the same issue in the car (Volvo XC40)...

**Comment by frankkienl** [comment]
You could try to file the issue here: 
https://issuetracker.google.com/issues?q=componentid:460472%20status:open#
That may help.

**Comment by TIB28** [comment]
Hi @nikagersonlohman , got the same problem, did you fix it ?

**Comment by nikagersonlohman** [comment]
No, sorry, I abandoned it because they denied testing the app further in my car as it's not a media app which is the only allowed currently :(

**Comment by andrewmbenton** [comment]
I know this is slightly off-topic but I saw that you did manage to install your testing app on an XC40 P8 via the Play Store and am wondering how you did that? I've set up testing channels for other apps and I think I did this one correctly (release type is Automotive OS). On the vehicle I'm signed in to the Google Account of one of my opted-in testers. But searching the Play Store on the vehicle turns up nothing. If you managed to do this successfully can you share the method?

**Comment by nikagersonlohman** [comment]
Hi Andrew,

I think it must have been a mistake. When I submitted the app a second time with hardly any changes, they denied it into the test phase...

This is what I got in februari:
![image](https://user-images.githubusercontent.com/75429776/123470559-dfe51180-d5f4-11eb-86b2-f55af73fbdd3.png)

And this is what I got last month:
![image](https://user-images.githubusercontent.com/75429776/123470644-fab78600-d5f4-11eb-96f8-4e37ee1eb71b.png)

Regards,

Nika.

**Comment by andrewmbenton** [comment]
My app has been approved for testing but yes presumably they will reject it at some point.

During the brief period when your app was approved for testing did you manage to run it on the car?

**Comment by nikagl** [comment]
Yes, and it's still running in it... 

🤷‍♂️

**Comment by andrewmbenton** [comment]
How did you manage to install? On the vehicle did you just search the Play Store from a tester's google account?

**Comment by SergeyStepanov** [comment]
Hi. I realize that my answer is already late, but I ran into this problem myself and I was able to solve it. So I am writing for those who will encounter it.

It is not enough to specify the permissions in the manifest. You need to request them in the GUI. I used the following example:

```
private val permissions = arrayOf(Car.PERMISSION_SPEED)

override fun onResume() {
        super.onResume()
        if(checkSelfPermission(permissions[0]) == PackageManager.PERMISSION_GRANTED) {
            //your code here
        } else {
            requestPermissions(permissions, 0)
        }
    }
```

**Comment by nikagersonlohman** [comment]
> How did you manage to install? On the vehicle did you just search the Play Store from a tester's google account?

It's a bit too long ago, I do not remember exactly, but I do remember it didn't show up initially. I had to setup beta in Google Play too. If I remember correctly I clicked like 10 times on the play store version and it said I am a developer (like developer options in android, which doesn't work in the car; it did work here). Here's a screenshot of what I needed to enable (translated it says: internal share apps on google play, download and install test-versions of apps shared with you):
![image](https://user-images.githubusercontent.com/75429776/123594498-12a72980-d7f0-11eb-8483-361e43e77f34.png)

After that, it didn't work initially. I could find the app but couldn't install it:
![image](https://user-images.githubusercontent.com/75429776/123594563-26eb2680-d7f0-11eb-8fa0-7fd0e15f9842.png)
![image](https://user-images.githubusercontent.com/75429776/123594545-20f54580-d7f0-11eb-8609-add6bb700f3b.png)

But after a while, it suddenly installed and is still running (even when driving):
![IMG_20210628_083618 (Large)](https://user-images.githubusercontent.com/75429776/123594739-5ef26980-d7f0-11eb-82be-c2afccc2df4f.jpg)
![IMG_20210628_083628 (Large)](https://user-images.githubusercontent.com/75429776/123594761-6580e100-d7f0-11eb-9b70-9d84a4b77258.jpg)
![IMG_20210628_084434 (Large)](https://user-images.githubusercontent.com/75429776/123594766-69146800-d7f0-11eb-85fa-1958468c2394.jpg)
![IMG_20210628_084524 (Large)](https://user-images.githubusercontent.com/75429776/123594784-6b76c200-d7f0-11eb-9868-d1748328caa1.jpg)
![IMG_20210628_084531 (Large)](https://user-images.githubusercontent.com/75429776/123594802-6fa2df80-d7f0-11eb-9402-9cdca9754b99.jpg)
![IMG_20210628_084538 (Large)](https://user-images.githubusercontent.com/75429776/123594819-73cefd00-d7f0-11eb-9db7-01e8a01a1be2.jpg)
![IMG_20210628_084551 (Large)](https://user-images.githubusercontent.com/75429776/123594839-77628400-d7f0-11eb-9cdd-6c7d98299828.jpg)
![IMG_20210628_084612 (Large)](https://user-images.githubusercontent.com/75429776/123594851-7a5d7480-d7f0-11eb-80d6-8b78b8d1cd7b.jpg)

**Comment by nikagersonlohman** [comment]
> Hi. I realize that my answer is already late, but I ran into this problem myself and I was able to solve it. So I am writing for those who will encounter it.
> 
> It is not enough to specify the permissions in the manifest. You need to request them in the GUI. I used the following example:
> 
> ```
> private val permissions = arrayOf(Car.PERMISSION_SPEED)
> 
> override fun onResume() {
>         super.onResume()
>         if(checkSelfPermission(permissions[0]) == PackageManager.PERMISSION_GRANTED) {
>             //your code here
>         } else {
>             requestPermissions(permissions, 0)
>         }
>     }
> ```

Superb! I'll try that soon on the emulator (as I can't get the app to publish to the car anymore :( )

**Comment by narko** [comment]
In case it helps, this file contains all the Car properties together with the required permissions:
https://cs.android.com/android/platform/superproject/+/master:packages/services/Car/car-lib/src/android/car/VehiclePropertyIds.java;l=30?q=VehiclePropertyIds
